tools/crm_mon.c

```diff
@@ -672,7 +672,7 @@ reconnect_after_timeout(gpointer data)
 static void
 mon_cib_connection_destroy(gpointer user_data)
 {
-    out->info(out, "Connection to the cluster-daemons terminated");
+    out->transient(out, "\nConnection to the cluster lost");
```

Comment:
Why a newline?

Reply:
Just because the string of ellipses (of varying length) before the first "Connection lost" message always bugged me:
```
Cluster Summary:
  * Stack: corosync
  * Current DC:	node1 (version 2.1.5-0b3656e85) - partition with quorum
  * Last updated: Thu Nov  3 17:02:27 2022
  * Last change:  Thu Nov  3 15:59:10 2022 by hacluster via crmd on node1
  * 1 node configured
  * 2 resource instances configured (1 DISABLED)

Node List:
  * Online: [ node1 ]

Active Resources:
  * dummy	(ocf:pacemaker:Dummy):   Started node1
......Connection to the cluster-daemons terminated
Connection to the cluster-daemons terminated
```

I'm not tied to the newline though. Other options besides the newline:
* Leave it as-is
* Modify the curses error and info/transient functions so that they move the cursor to the beginning of the current line before printing. Not sure if that would have any unintended consequences, but it definitely looks the nicest in this scenario:
```
Cluster Summary:
...
Active Resources:
  * dummy	(ocf:pacemaker:Dummy):   Started node1
Connection to the cluster lost
Connection to the cluster lost
```

Reply:
I definitely think the handling should be in the implementation rather than the message string. The idea is that the message string should be identical across all formats and not have format-specific control characters.

Moving to the beginning of the line should be fine for error at least (clearing the line would also be a good idea in case the existing content is longer than the error message). For info/transient it would probably be fine too. I suppose the ideal would be to check the line position and output a newline if not at the beginning, but that might not be worth the hassle.

Reply:
I'm going to just remove the newline from the commit for now. We can revisit this later.

> that might not be worth the hassle.

For curses, doing something based on the current position is no more difficult than moving to the beginning of the line. Which is to say, not very. It's probably doable in other formats as well (something like get the character at `ftell(fd) - 1` and act if that character is not a newline? Haven't done much with streams).

There may or may not be existing places where we build part of a string piece-by-piece based on conditions, and then finish it with `out->info(out, "the rest")`. Any new approach that adds a newline or overwrites part of a line would break that.

Reply:
There shouldn't be, we rejected that possibility when designing the interface. We use the usual techniques to build a string then pass the whole string to the output object function. That's the only way to abstract e.g. text vs xml needing a newline or not. We considered separate methods for build-up-a-line and end-a-line but there are already functions for that purpose, and not all formats are line-oriented.

Just dropping the newline is fine for this PR